lib/data/moor_database.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:moor/moor.dart';
+
+part 'moor_database.g.dart';
+
+@DataClassName("CalendarItem")
+class CalendarItems extends Table {
+  //autoincrement automatically sets id as the primary key
+  IntColumn get id => integer().autoIncrement()();
+
+  TextColumn get title => text().withLength(min: 1)();
```

Comment:
Does .withLength(min:1) ensure that the title field is not empty? When you implement addCalendarEvent, you will perform these checks before even trying to add.

Reply:
It'll check the title field whenever it is updated or a value is inserted and if the value < min length then it'll throw an exception and no data will actually be changed. If we're going to perform the checks when adding anyways then this method shouldn't matter